new-lamassu-admin/src/pages/Blacklist/Blacklist.js

```diff
@@ -0,0 +1,183 @@
+import { useQuery, useMutation } from '@apollo/react-hooks'
+import { Box } from '@material-ui/core'
+import Grid from '@material-ui/core/Grid'
+import { makeStyles } from '@material-ui/core/styles'
+import gql from 'graphql-tag'
+import * as R from 'ramda'
+import React, { useState } from 'react'
+
+import Tooltip from 'src/components/Tooltip'
+import { Link } from 'src/components/buttons'
+import { Switch } from 'src/components/inputs'
+import Sidebar from 'src/components/layout/Sidebar'
+import TitleSection from 'src/components/layout/TitleSection'
+import { H4, Label2, P } from 'src/components/typography'
+import { fromNamespace, toNamespace } from 'src/utils/config'
+
+import styles from './Blacklist.styles'
+import BlackListModal from './BlacklistModal'
+import BlacklistTable from './BlacklistTable'
+
+const useStyles = makeStyles(styles)
+
+const groupByCode = R.groupBy(obj => obj.cryptoCode)
+
+const DELETE_ROW = gql`
+  mutation DeleteBlacklistRow($cryptoCode: String!, $address: String!) {
+    deleteBlacklistRow(cryptoCode: $cryptoCode, address: $address) {
+      cryptoCode
+      address
+    }
+  }
+`
+
+const GET_BLACKLIST = gql`
+  query getBlacklistData {
+    blacklist {
+      cryptoCode
+      address
+    }
+    cryptoCurrencies {
+      display
+      code
+    }
+  }
+`
+
+const SAVE_CONFIG = gql`
+  mutation Save($config: JSONObject) {
+    saveConfig(config: $config)
+  }
+`
+
+const GET_INFO = gql`
+  query getData {
+    config
+  }
+`
+
+const ADD_ROW = gql`
+  mutation InsertBlacklistRow($cryptoCode: String!, $address: String!) {
+    insertBlacklistRow(cryptoCode: $cryptoCode, address: $address) {
+      cryptoCode
+      address
+    }
+  }
+`
+
+const Blacklist = () => {
+  const { data: blacklistResponse } = useQuery(GET_BLACKLIST)
+  const { data: configData } = useQuery(GET_INFO)
+  const [showModal, setShowModal] = useState(false)
+  const [clickedItem, setClickedItem] = useState({
+    code: 'BTC',
+    display: 'Bitcoin'
+  })
```

Comment:
Easier to just set the code as state, no reason to have a object here.

Reply:
as we talked in private message, we do need this after all 👍 :)